### PR TITLE
#minor Intratask checkpointing in FlytePropeller

### PR DESCRIPTION
**Pull request by kumare3**
Signed-off-by: Ketan Umare <ketan.umare@gmail.com>

# TL;DR
This uses the PR - https://github.com/flyteorg/flyteplugins/pull/221, to provide intra-task checkpointing support. It depends on the Flyteplugins PR so should not be checked in prior to 221

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
Flytepropeller simply uses the already existing raw-output-prefixes and generates a subpath that can house a Flyte aware checkpoint and is passed between consecutive retries.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/643

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/controller/nodes/task/taskexec_context_test.go [hunk 1]
```diff
@@ -340,3 +351,70 @@ func TestConvertTaskResourcesToRequirements(t *testing.T) {
...
+		},
+		promutils.NewTestScope(),
+	)
+	assert.NoError(t, err)
+	nCtx.OnDataStore().Return(ds)
+	nCtx.OnNodeExecutionMetadata().Return(nm)
+
+	pre, uid, err := ComputeRawOutputPrefix(context.TODO(), 100, nCtx, "n1", 0)
+	assert.NoError(t, err)
+	assert.Equal(t, uid, "name-n1-0")
```
Comment (hamersaw) [inline pkg/controller/nodes/task/taskexec_context_test.go:374]:
```suggestion
	assert.Equal(t, "name-n1-0", uid)
```
change order of args because test output uses first for "expected" and second for "observed"

File: pkg/controller/nodes/task/taskexec_context_test.go [hunk 2]
```diff
@@ -340,3 +351,70 @@ func TestConvertTaskResourcesToRequirements(t *testing.T) {
...
+	assert.NotNil(t, pre)
+	assert.Equal(t, pre.GetRawOutputPrefix(), storage.DataReference("s3://sandbox/x/name-n1-0"))
```
Comment (hamersaw) [inline pkg/controller/nodes/task/taskexec_context_test.go:376]:
```suggestion
	assert.Equal(t, storage.DataReference("s3://sandbox/x/name-n1-0"), pre.GetRawOutputPrefix())
```

File: pkg/controller/nodes/task/taskexec_context_test.go (hunk 1 above)
Comment (kumare3) [reply pkg/controller/nodes/task/taskexec_context_test.go:374]: Yup

File: pkg/controller/nodes/task/taskexec_context_test.go (hunk 2 above)
Comment (kumare3) [reply pkg/controller/nodes/task/taskexec_context_test.go:376]: Will do

**Review by hamersaw — APPROVED** [review]
